File: Algorithms/Sorting/counting_sort.c
#include <stdio.h>

void
counting_sort (int* seq, int n)
{
  int cnt[256] = {0};

  for (int i = 0; i < n; ++i)
    cnt[seq[i]] += 1;

  int k = 0;
  for (int i = 0; i < 256; ++i)
    for (int j = cnt[i]; j > 0; --j)
      seq[k++] = i;
}


int
main ()
{
  int seq[] = {3, 5, 2, 4, 1};
  int n = 5;

  counting_sort (seq, n);

  for (int i = 0; i < n; ++i)
    printf ("%d ", seq[i]);
  printf ("\n");
}

File: Algorithms/Sorting/counting_sort.py
#!/usr/bin/python3

def counting_sort(seq):
    cnt = [0] * 256

    for i in range(len(seq)):
        cnt[seq[i]] += 1

    k = 0
    for i in range(256):
        j = cnt[i]
        while j > 0:
            seq[k] = i
            k += 1
            j -= 1

    return seq

seq = [3, 5, 2, 4, 1]
print(counting_sort(seq))
